### ISSUE TITLE
Slicing on non uniform array

### ISSUE THREAD

**Issue by ykjawale** [numpy/numpy#11687]
slicing does not work with non-unifrom arrays (eg. b)
```python
import numpy as np
a = np.array([[1,2,3],[4,5,6],[7,8,9]])
a[1,:]
b = np.array([[1,2,3,4],[5,6],[7,8,9]])
b[1,:]
```
Can ```:``` element be implemented as first-to-last and not indexing based

**Comment by seberg** [comment]
The thing is that numpy has no notion of "non-uniform" arrays, all arrays are uniform. So the second operation is not really valid from a numpy perspective. If you look closely `b` is a one dimensional arrays containing python lists!

Since numpy is inherently only uniform arrays, I do not think there is anything that can be done here. Note if you just have a colon, you can also write e.g. an Ellipsis (to just take all following dimensions). Otherwise I would have to point you to another tool that might support such ragged arrays. I am not 100% sure about those from the top of my head, possibly `dynd` though I am not sure what their status is.

So unfortunately, have to close this as out of scope for numpy. If you have questions on other solutions or so, feel free to continue discussion.

**Comment by eric-wieser** [comment]
I think we should consider having `np.array([[1,2,3,4],[5,6],[7,8,9]])` emit a warning - it's a trivial change after #11601

**Comment by seberg** [comment]
Ah, that PR is nice. Since it allows more non-failure cases (I guess) maybe it would be an especially good time to add that warning indeed.
